### ISSUE TITLE
Use JSON Schema instead of prose descriptions.

### ISSUE THREAD

**Issue by duesee** [mlswg/mls-implementations#83]
We could replace the test case descriptions with [JSON Schema](https://json-schema.org/) and check the test vectors in CI. This way, it cannot happen that things get out-of-sync. I could give it a try when we want it :-)

**Comment by bifurcation** [comment]
It's a nice idea, but I don't think anyone will actually use this in practice.  I would note that there are also semantics here that will not be captured by JSON Schema, e.g., the fact that a string represents a hex-encoded, TLS-serialized struct of a certain type.

Might be useful as a finishing touch once we have an agreed-upon set of test vectors, but probably not during development.

**Comment by duesee** [comment]
Yeah, maybe. The idea was also to agree on how to document the test vectors in a consistent way. I saw JSON Schema used for this in [Project Wycheproof](https://github.com/google/wycheproof/blob/master/schemas/aead_test_schema.json) and found it to be very helpful. But I agree that it's propbably too early to think about this.

Regarding the hex-semantics: Wycheproof uses, e.g., ...

```json
        "key": {
          "type": "string",
          "format": "HexBytes",
          "description": "the key"
        },
```

... to define a hex-encoded key. There are some pre-defined `format`s in JSON Scheme but AFAIK "HexBytes" is none of them. So I'm not sure a valiadtion tool could validate that.
